### PR TITLE
Add options for "Save as" and "Download" buttons in Wodo.TextEditor

### PR DESCRIPTION
**Pull request by kossebau**
These buttons are used in the deployment of Wodo.TextEditor in https://github.com/Pleio/elgg-odt_editor

Given the not surprising usecase (editing documents stored on a server) they might be useful in similar deployments, so it might make sense to have these patches (done by me) also upstream.

Will screw a little once more the started port to Closure widgets, but well...

### CODE REVIEW

File: programs/editor/Tools.js [hunk 1]
```diff
@@ -183,6 +186,36 @@ define("webodf/editor/Tools", [
                     saveButton.placeAt(toolbar);
                 }
 
+                // SaveAs
+                if (saveAsOdtFile) {
```
Comment (peitschie) [inline programs/editor/Tools.js]: This stuff really points towards the need to make this toolbar properly extensible somehow. I'd expect the first thing a certain type of 3rd party will want to do is to switch to their own chrome instead.

Oh well... it's following what the current Romans have already done with this file :smiling_imp:

Comment (kossebau) [inline programs/editor/Tools.js]: Yes, proper extensibility would be good to have. But for now lower priority than the Closure port (for me). Once that port is done, extensibility (along plugin support) should be gone for.

Besides that general comment... did I see a ship-it hidden between the lines? :angel:

**Comment by peitschie** [comment]
My apologies... we're cutting a release, and things got a little warm in the frying pan after my first comment.

Please, release this patch into the pool of commits that make up webodf master!

**Comment by kossebau** [comment]
Thanks (so you are crispy now? :) )

**Pull request merged**